File: functions/model/FortuneUser/components/UserAiFortune.js
class UserAiFortune {
    constructor({ fortuneText, fortuneDate, fortuneNo }) {
        this.fortuneText = fortuneText,
            this.fortuneDate = fortuneDate,
            this.fortuneNo = fortuneNo
    }

    static fromJson(json) {
        return new UserAiFortune({
            fortuneText: json["fortuneText"],
            fortuneDate: json["fortuneDate"],
            fortuneNo: json["fortuneNo"]
        });
    }

    toJson() {
        return {
            "fortuneText": this.fortuneText,
            "fortuneDate": this.fortuneDate,
            "fortuneNo": this.fortuneNo
        };
    }
}
module.exports = { UserAiFortune };

File: lib/src/model/FortuneUser/components/user_ai_fortune.dart
import 'package:cloud_firestore/cloud_firestore.dart';

class UserAiFortune {
  String fortuneText;
  Timestamp fortuneDate;
  String fortuneNo;

  UserAiFortune({
    required this.fortuneText,
    required this.fortuneDate,
    required this.fortuneNo,
  });

  factory UserAiFortune.fromJson(Map<String, dynamic> json) => UserAiFortune(
        fortuneText: json["fortuneText"],
        fortuneDate: json["fortuneDate"] as Timestamp,
        fortuneNo: json["fortuneNo"],
      );

  Map<String, dynamic> toJson() => {
        "fortuneText": fortuneText,
        "fortuneDate": fortuneDate,
        "fortuneNo": fortuneNo,
      };
}